### PR TITLE
Pabu

### PR DESCRIPTION
**Pull request by quoll**
Issue #20: Allows for user interaction via a cli and Datalog

### CODE REVIEW

File: src/naga/lang/pabu.clj [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      :author "Paula Gearon"}
+  naga.lang.pabu
+  (:require [naga.schema.structs :as structs :refer [Axiom Program]]
+            [naga.lang.parser :as parser]
+            [naga.rules :as r]
+            [schema.core :as s])
+  (:import [java.io InputStream]
+           [naga.schema.structs Rule]))
+
+;; TODO: Multi-arity not yet supported
```
Comment (saintx) [inline src/naga/lang/pabu.clj:12]: Why not?  Does this need its own issue with an "improvement" label?

File: src/naga/lang/pabu.clj [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  (if (= 1 (count args))
+    [s :rdf/type property]
+    [s property o]))
+
+(s/defn ast->axiom :- Axiom
+  "Converts the axiom structure returned from the parser"
+  [{axiom :axiom :as axiom-ast} :- AxiomAST]
+  (triplet axiom))
+
+(def VK (s/cond-pre s/Keyword s/Symbol))
```
Comment (saintx) [inline src/naga/lang/pabu.clj]: docstring

File: src/naga/lang/pabu.clj (hunk 1 above)
Comment (quoll) [reply src/naga/lang/pabu.clj:12]: Not done yet, as it is not necessary for a minimal implementation, and contains some complexity.
Added issues #21 and #22 for this feature.

**Comment by saintx** [comment]
This looks good to me, based on a hasty initial reading, but I would definitely need more time to get a fuller sense of how it works.  Let's get it merged, and continue to flesh out the tests and examples.

**Pull request merged**